exercises/GameOfLife/typescript/script.ts

```diff
@@ -0,0 +1,184 @@
+const canvas = document.querySelector<HTMLCanvasElement>("#game");
+const width = canvas.width;
+const height = canvas.height;
+const ctx = canvas.getContext("2d");
+const TILE_SIZE = 20;
+const TILES_X = width / TILE_SIZE;
+const TILES_Y = height / TILE_SIZE;
+ctx.fillStyle = "rgb(100, 240, 150)";
+ctx.strokeStyle = "rgb(90, 90, 90)";
+ctx.lineWidth = 1;
+
+const drawBorders = () => {
+    for(let i=0;i<TILES_X;i++) {
+        ctx.beginPath();
+        ctx.moveTo(i * TILE_SIZE - 0.5, 0);
+        ctx.lineTo(i * TILE_SIZE - 0.5, height);
+        ctx.stroke();
+    }
+for(let j=0;j<TILES_Y;j++) {
+        ctx.beginPath();
+        ctx.moveTo(0, j * TILE_SIZE - 0.5);
+        ctx.lineTo(width, j * TILE_SIZE - 0.5);
+        ctx.stroke();
+    }
+}
+
+const prepareBoard = (): boolean[][] => {
+    const b = [];
+    for(let i=0;i<TILES_X;i++) {
+        const row = [];
+        for(let j=0;j<TILES_Y;j++) {
+            row.push(false);
+        }
+        b.push(row);
+    }
+    return b;
+}
+
+let BOARD = prepareBoard();
+BOARD[1][0] = true;
+BOARD[2][1] = true;
+BOARD[0][2] = true;
+BOARD[1][2] = true;
+BOARD[2][2] = true;
+
+const isAlive = (x: number, y: number): number => {
+    if (x < 0 || x >= TILES_X || y < 0 || y >= TILES_Y) {
+        return 0;
+    }
+    return BOARD[x][y] ? 1 : 0;
+}
+const neighboursCount = (x: number, y: number): number => {
+    let count = 0;
+            /*if ((x == 0) || (x == TILES_X-1)){
+
+                for(let j of [-1,0,1]){
+
+                    count += isAlive(TILES_X-1-x,TILES_X-1-y+j);
+                }
+            }
+            else if ((y == 0) || (y == TILES_Y-1)){
+
+                for(let i of [-1,0,1]){
+
+                    count += isAlive(TILES_X-1-x+i,TILES_Y-1-y);
+                }
+            }*/
```

Comment:
I know I mentioned this to you in person but now that I'm actually looking at the code I _STRONGLY_ encourage you use mod (`%`) here instead of writing special code for the edges.